### PR TITLE
Allow implicit cast bool to Field

### PR DESCRIPTION
**Pull request by vitlibar**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by vitlibar** [comment]
@novikd Do you have any notes or suggestions?

File: src/Core/Field.h [hunk 1]
```diff
@@ -345,7 +346,9 @@ class Field
     }
 
     template <typename T>
-    Field(T && rhs, enable_if_not_field_or_stringlike_t<T> = nullptr);
+    Field(T && rhs, enable_if_not_field_or_bool_or_stringlike_t<T> = nullptr);
+
+    Field(bool rhs);
```
Comment (novikd) [inline src/Core/Field.h:351]: I'd prefer to implement this constructor right here because it's just a one-liner.

**Review by novikd — APPROVED** [review]
LGTM

**Comment by vitlibar** [comment]
Issues with integration tests and testflows will be fixed in https://github.com/ClickHouse/ClickHouse/pull/28063

**Review by alesapin — COMMENTED** [review]
Why do we need it? Implicit casts to such structure as `Field` can lead to many subtle bugs.

**Comment by vitlibar** [comment]
> Why do we need it? Implicit casts to such structure as `Field` can lead to many subtle bugs.

For every other class I'd agree, but `Field` is a very special class because it's designed to keep values of different types. And we already have lots of implicit casts `signed char -> Field`, `UInt16 -> Field`, `Int32 -> Field`, `Float64 -> Field`, `std::string_view -> Field`, and many others, so I think it's ok.

**Pull request merged**